include/bluetooth/conn.h

```diff
@@ -184,6 +184,14 @@ struct bt_conn_info {
 	};
 };
 
+/** @brief Check if connected
+ *
+ *  @param conn Connection object
+ *
+ *  @return true if connected, false if disconnected.
+ */
+bool bt_conn_is_connected(const struct bt_conn *conn);
```

Comment:
How about we return an `int` and rename this to `bt_conn_get_state()`. I know that sooner or later we will want more info. For example, if it's disconnecting or if it's currently encrypting or whatever you can think of.

Reply:
Then we'd need to expose the internal enum and document it (i.e. the return would be the enum instead of an int). It's actually a typedef internally (in `conn_internal.h`): `bt_conn_state_t`

Reply:
@jhedberg @carlescufi 
I think it would be better to expose a reduced state in order to avoid having to deal with internal specific states (like SCAN CONNECT ADV_DIR).
Create an exposed state like:
CONNECTING
CONNECTED
DISCONNECTING
DISCONNECTED

Also to avoid cluttering the function space we could add this state to the bt_conn_info object?

Encrypted/security state is available through the API bt_conn_get_security()

Reply:
TBH I'm in general not sure about adding this as a public API since it's inherently racy. Application side code will often run in preemptive threads, so the state of a connection may change between checking it and trying to do something with a connection. The missing checks for GATT still need to be fixed of course - might have been better to have those in a separate PR since they're less contentious.

Reply:
Yeah, I was unsure about including it in the PR, and since this is turning into a bigger discussion I'll remove it for now so that we can get the GATT fixes.
I'll create a new PR (if needed) for the connection state.